### PR TITLE
Do not report document metering on system indices

### PR DESCRIPTION
**Pull request by pgomulka**
For system indices we don't want to emit metrics. DocumentSizeReporter will be created given an index. It will internally contain a SystemIndices instance that will verify the indexName with `isSystemName`

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/part-1

File: server/src/main/java/org/elasticsearch/ingest/IngestService.java [hunk 1]
```diff
@@ -740,7 +740,9 @@ protected void doRun() {
                         totalMetrics.preIngest();
                         final int slot = i;
                         final Releasable ref = refs.acquire();
-                        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver();
+                        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver(
+                            indexRequest.index()
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/ingest/IngestService.java:744]: Ingest processors can change the index name, right? So then do we need this to be lazily set? Perhaps a method on the observer itself?

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/ingest/IngestService.java:744]: I can make this lazy - set after ingestpipelines, but I suspect this is rare and would mean we are executing metering logic when we don't need it?
Requests to system indices cannot originate from regular users, and also regular users should not be able to reroute their indexing requests to system indices. Although I cannot see any validation for this..

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**